docs/feature_rgb_matrix.md

```diff
@@ -369,7 +369,7 @@ These are defined in [`rgblight_list.h`](https://github.com/qmk/qmk_firmware/blo
 ```C
 #define RGB_MATRIX_KEYPRESSES // reacts to keypresses
 #define RGB_MATRIX_KEYRELEASES // reacts to keyreleases (instead of keypresses)
-#define RGB_DISABLE_AFTER_TIMEOUT 0 // number of ticks to wait until disabling effects
+#define RGB_DISABLE_AFTER_TIMEOUT 0 // number of milliseconds to wait until rgb automatically turns off
```

Comment:
Given the behaviour of `RGB_DISABLE_AFTER_TIMEOUT` has now changed by a factor of `1000 / 1.2`, I can see 2 ways forward:
1. Go through the breaking change process so it merges during a documented and expected period of change.
2. Add a new config option something like `RGB_DISABLE_TIMEOUT` which is milliseconds based, and some backwards compatibility of `RGB_DISABLE_AFTER_TIMEOUT /  (1000 * 1.2)` (or whatever scaling factor is required).

Reply:
Lets just pull it into the future branch that is coming up if it is not too late. 

Also the original behavior has been broken for white a while. As it was once  tick based with a 20hz tick rate which changed with the huge rgb matrix overhaul to where it is now 1.2s based. Also as a note, there is not a single keyboard or keymap in qmk today that even uses this define. Not to say it's completely unused as folks could be using it locally.

Reply:
Future is closed to new PRs at this time. It's too late to merge it in there. Might need to mark this PR as `on hold`.

Reply:
Well, up to you all if you want to wait a full quarter until rgb matrix timeout is fixed.

Reply:
I don't personally know why this is a breaking change in @zvecr's opinion, but I'm not familiar enough with the RGB codebase to determine that on my own (I only have one board with RGB underglow, and it's not assembled/completed). I think his concern may be that users that don't have their keymaps in qmk:master may have unexpected behaviour from this change, and we have no way of knowing if that's the case. That's the only thing I can think of.

As qmk:master is going to be frozen in a couple days, I don't think there's adequate time to put this in future(_unformatted) and test it properly, in which case we may need to wait until after qmk:master is unfrozen to bring this in. That's the only thing I meant to convey above.

Reply:
@noroadsleft Thats correct.

(assumptions made on the code comments) If a user previously set `RGB_DISABLE_AFTER_TIMEOUT` to 100, it would be (100 * 1.2s) so 2 mins. Now if they upgrade, the timeout is in milis and would time out after 100ms. While not a massive deal, it is a change in behaviour to which they have no direct notice of, thus I flagged it.

Reply:
I may be biased here, but I think it should be okay to merge as is. 

But part of that is because I don't know why anyone would want to turn off the RGB LEDs, other than when the host system is off/asleep.  (hence my bias).

Reply:
I don't think its safe to say its okay to merge this as you would personally use a different feature of the firmware.

What needs to be decided is, is this particular behaviour change acceptable without any mitigation code. A user gets no knowledge if they update their repo, that a feature they were using has changed. As this is a rolling release process, you would have to join the dots from the commit history, documentation, and keymap. Remember, some users are just copy pasting suggestions that they get from discord/reddit and only doing so as the configurator cannot meet their needs.

Reply:
Usually how it is handled in the software development world:
- Breaking change: changes that prevents any possible configuration from compiling.
- Functional change: changes that don't break compiling, but could affect the output considerably.
- Bug fix: just bug fix.

Breaking changes are usually delayed until a major version release, such as the breaking change label in QMKs case. Functional change, is usually delayed until a minor version release, and patch fixes are usually just pushed out with the next revision version. Bug fixes can also fall into the functional change category as they can affect code behavior that users were relying upon. Depending on the number of affected users, a bugfix may be reverted to restore previous behavior if the demand is great enough.

In the case of this PR, there is a functional change as well as a patch change. The impact of the functional change can be determined by examining the existing keymaps, and how many times the reported issue came up. In which case, the impact is near 0 as no keymaps in qmk are affected, and no reports from users about this feature not working. It cannot fully be 0 as there could be cases out there we are not aware of, and that is part of the risk assessment that you all need to take into account on pr go or no go.

Reply:
So if I understand this correctly, this value was arbitrary to begin with and we've now made it based on a real world unit. The end affect, if anyone was using it their timeout is now about 20% shorter than they configured originally?

Reply:
it was 1.2s units before (arbitrary) and now 1ms units. So 1/1200th shorter than before.

Reply:
ah, I see. no-one in tree seems to be using it, this seems to inconvenience about 6 keyboards if people are keeping their keymaps out of tree. I think I agree with Option 2 from @zvecr choices. It seems like the right thing to do.

Reply:
@noroadsleft @yanfali @zvecr Fixed, marked old define as obsolete with an automatic conversion path forward.